src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -334,7 +335,7 @@ internal override TypeWithAnnotations IteratorElementTypeWithAnnotations
 
         public SyntaxToken NameToken => Syntax.Identifier;
 
-        internal override Binder SignatureBinder => _binder;
+        internal override Binder SignatureBinder => _parameterBinder;
```

Comment:
>SignatureBinder [](http://example.com/codeflow?start=33&length=15)

I find the current terminology (```SignatureBinder```, ```ParameterBinder```) very confusing an it looks like this could be the source of having the problem that we are trying to address in the first place. Also, it looks like there is an inconsistency in the way different symbols implement ```ParameterBinder```. For example, ```LambdaSymbol``` implements it this way:
```
internal override Binder ParameterBinder => new WithLambdaParametersBinder(this, _binder);
```
I.e. the parameters are in scope, but for ```LocalFunctionSymbol``` the parameters are not in scope. There could be other inconsistencies as well.
I think we should switch to the terminology that clearly reflects what do we expect to have in scope. By looking at ```BinderFactoryVisitor.VisitMethodDeclaration```, we might need at most tree binders:
- outer scope binder (method itself is not involved)
- with type parameters binder
- with method parameters binder

Let's figure out which binders we actually need and add specific APIs for them, replacing the old properties and adjusting all consumers as appropriate.